src/cmd/ksh93/tests/builtins.sh

```diff
@@ -476,15 +476,14 @@ $SHELL <<- \EOF
     cd
     [[ $(pwd) == "$home" ]]
 EOF
-if [[ $? != 0 ]]
+ret=$?
+if [[ ! -d ~$USER ]]
```

Comment:
Better but please invert the logic; i.e., only do this test if the previous command failed since there isn't any point in doing this if it succeeded. For example,

```
if [[ $? != 0 ]]
then
    if [[ -d ~$USER ]]
    then
        log_error 'cd with no arguments fails if HOME is unset'
    else
        log_warning "ignoring test of cd with no arguments if home directory does not exist"
    fi
fi
```